File: sdk/ts/examples/workflow.ts
import { newWorkflow, node, runWorkflow, onBranch } from "../src";

const wf = newWorkflow("build-test-deploy", "v1.2.0",
  node("lint"),
  node("test").dependsOn("lint"),
  node("build").dependsOn("test"),
  node("deploy").dependsOn("build").when(onBranch("main")),
);

runWorkflow(wf);

File: sdk/go/examples/workflow.go
//go:build ignore

package main

import brazier "github.com/brazier/sdk/go"

func main() {
	wf := brazier.NewWorkflow("build-test-deploy", "v1.2.0",
		brazier.WorkflowNodes(
			brazier.Node("lint"),
			brazier.Node("test").DependsOn("lint"),
			brazier.Node("build").DependsOn("test"),
			brazier.Node("deploy").DependsOn("build").When(brazier.OnBranch("main")),
		),
	)

	brazier.RunWorkflow(wf)
}
